.travis.yml

```diff
@@ -91,160 +53,150 @@ matrix:
-          env: SETUP_CMD='test --remote-data=astropy --readonly'
-               LC_CTYPE=C.ascii LC_ALL=C
-               PIP_DEPENDENCIES="" CONDA_DEPENDENCIES=""
-               INSTALL_WITH_PIP=True
-               EXTRAS_INSTALL="test,all"
 
-        - os: linux
+        # For the Numpy 1.16 build we use oldestdeps not numpy116 since we can check
+        # the oldest version of all dependencies where this is known.
+        - language: python
+          python: 3.6
+          env: TOXENV="test-oldestdeps"
+               TOXPOSARGS="--open-files"
+
+        # Now try with all optional dependencies.
+
+        - language: python
+          python: 3.6
           stage: Initial tests
-          env: PYTHON_VERSION=3.7 CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
-               PIP_DEPENDENCIES="scikit-image $ASDF_PIP_DEP"
-               SETUP_CMD='test -a "--durations=50"'
+          env: TOXENV="test-alldeps"
+               TOXPOSARGS="--durations=50""
           compiler: clang
 
         # Full tests with coverage checks.
-        - os: linux
-          env: SETUP_CMD='test --coverage --remote-data=astropy --readonly'
-               CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
-               PIP_DEPENDENCIES="scikit-image codecov objgraph jplephem bintrees sortedcontainers $ASDF_PIP_DEP"
+        - language: python
+          python: 3.7
+          env: TOXENV="test-alldeps"
+               TOXPOSARGS="--remote-data=astropy"
+               COVERAGE=True
                LC_CTYPE=C.ascii LC_ALL=C
                CFLAGS='--coverage -fno-inline-functions -O0'
-               MATPLOTLIB_VERSION=2.1
                EVENT_TYPE='push pull_request cron'
 
         # Try on Windows
         - os: windows
           stage: Final tests
-          env: PYTHON_VERSION=3.7 SETUP_CMD='test --readonly'
-               CONDA_DEPENDENCIES=""
-               PIP_DEPENDENCIES="Cython scipy h5py beautifulsoup4 html5lib jinja2 pyyaml matplotlib scikit-image pytz pandas objgraph asdf"
+          env: PYTHON_VERSION=3.7
+               USE_CI_HELPERS=True
+               TOXENV='test-alldeps'
 
         # Do a PEP8/pyflakes test with flake8
-        - os: linux
+        - language: python
+          python: 3.7
           stage: Initial tests
-          env: MAIN_CMD="flake8 astropy --count $FLAKE8_OPT" SETUP_CMD=''
+          env: TOXENV=style
 
         # Try developer version of Numpy with optional dependencies and also
         # run all remote tests. Since both cases will be potentially
         # unstable, we combine them into a single unstable build that we can
         # mark as an allowed failure below.
-        - os: linux
+        - language: python
+          python: 3.7
           stage: Final tests
-          env: PYTHON_VERSION=3.7 NUMPY_VERSION=dev SETUP_CMD='test --remote-data'
-               CONDA_DEPENDENCIES=''
-               PIP_DEPENDENCIES=$DEV_PIP_DEP
-               MATPLOTLIB_VERSION=dev
+          env: TOXENV="test-devdeps"
```

Comment:
I'd do `test-dev-dependencies` (even I wouldn't go to `test-development-dependencies`...)